featuretools/tests/entityset_tests/test_ww_es.py

```diff
@@ -824,3 +824,30 @@ def test_normalize_ww_init():
 
     assert es['new_df'].ww.name == 'new_df'
     assert es['new_df'].ww.schema.name == 'new_df'
+
+
+def test_update_dataframe_schema():
+    before = pd.DataFrame({
+        'index': [0, 1, 2],
+        'boolean': [True, False, True],
+        'latlong': [(1.0, 2.0), (3.0, 4.0), np.nan],
+    })
+
+    before.ww.init(index='index', logical_types={
+        'boolean': ww.logical_types.Boolean,
+        'latlong': ww.logical_types.LatLong,
+    })
+
+    es = EntitySet()
+    es.add_dataframe(dataframe_name='data', df=before)
```

Comment:
I think this needs to be updated. If we are adding a Woodwork dataframe, it needs to have a `name` assigned. So we can either add a name in the `ww.init` call, or we can skip the init and specify the index and logical types during the add_dataframe call.

Also, the `df` parameter needs to be updated to `dataframe`